Guarded typing_extensions by version

ezragoss opened the pull request:
- Added version guards appropriate to typing_extensions
- Made typing_extensions a required import for versions 3.7-3.9.* to match the version guards
- Added poetry.lock to the gitignore

chadrik commented on typeright/annotations/main.py:
```diff
@@ -1,9 +1,13 @@
 
-from typing_extensions import TypedDict
+if sys.version_info.major >= 3 and 7 <= sys.version_info.minor < 10:
+    from typing_extensions import TypedDict
```
I find this more legible:

```python
if sys.version_info[:2] >= (3, 7) and sys.version_info[:2] < (3, 10):
```

chadrik commented on pyproject.toml:
```diff
@@ -31,20 +31,20 @@ six = "*"
 docutils = "*"
 "kids.cache" = "*"
 typing = { version = "3.7.*", python = "<3.7" }
+typing-extensions = { version = ">=3.7.4.3", python = ">=3.7,<3.10" }
```
I think we want to install typing-extensions for `python = "<3.10"`, since we'll need it in python2 as well. I confirmed that `3.7.4.3` is available for python2.   In which case, the imports will look like this:

```python
if sys.version_info[:2] < (3, 10):
    from typing_extensions import TypedDict
else:
    from typing import TypedDict
```

ezragoss replied:
Nice, that makes sense. I should have checked whether typing-extensions was py2 compliant!

I agree that your (major, minor) version is more legible. I can make the change and update the PR tonight.

chadrik reviewed (commented):
Hey Ezra, thanks for the contribution!  Are you guys using mypy / static-typing or is this for a personal project?

ezragoss commented:
We're trying to get typing more widely adopted across the studio but our small team at IW has been typing new code for a little over a year (largely influenced, at least on our team, by what I learned at Luma). Most folks at different game studios I've spoken to seem to be interested in adopting static typing and a big obstacle is being able to handle untyped legacy code in a linting/typechecking pre-commit workflow so that's why I'm revisiting typeright. Mostly it would be nice to have it in my back pocket as an auto-typing solution whether professional or personal.

Maybe for legal reasons I should say these are personal contributions on personal time haha

ezragoss committed:
Guarded import by version , fixed pre-submit

- Guarded typing_extensions import by version
- Updated isort in pre-commit hook to 5.12.0 because lower than that is incompatible with poetry per https://stackoverflow.com/questions/75269700/pre-commit-fails-to-install-isort-5-11-4-with-error-runtimeerror-the-poetry-co

ezragoss commented:
Ok I also added a pre-commit fix and linked the [relevant source](https://stackoverflow.com/questions/75269700/pre-commit-fails-to-install-isort-5-11-4-with-error-runtimeerror-the-poetry-co) for changing the isort version in pre-submit hooks. Let me know if you'd rather I break up those commits

chadrik commented:
Looks good, thanks!

You may also be interested in this new project that I'm working on:  https://github.com/LumaPictures/cg-stubs/

I'll be getting them all onto pypi by early next week.
